generator/internal/api/model.go

```diff
@@ -41,6 +46,56 @@ const (
 	SINT64_TYPE                 // 18
 )
 
+type Element interface {
+	Accept(v *Visitor) error
+}
+
+type Visitor interface {
+	VisitAPI(a *API) error
+	VisitMessage(m *Message) error
+	VisitService(s *Service) error
+	VisitMethod(m *Method) error
+	VisitMessageElement(m *MessageElement) error
+	VisitPathInfo(p *PathInfo) error
+	VisitPathSegment(s *PathSegment) error
+}
```

Comment:
Consider refactoring to its own file.

Reply:
I'll add a single `visitors.go` file to the api package for all the interfaces and the two visitor implementations I added.